emission/core/wrapper/suggestion_sys.py

```diff
@@ -360,7 +358,39 @@ def geojson_to_lat_lon_separated(geojson):
 
 '''
 REWRITE def check_mode_from_trip(cleaned_trip, cleaned_sections, section_counter, trip_counter): 
+Mode number correspondence: 
+0: "IN_VEHICLE"
+1: "BIKING"
+2: "ON_FOOT"
+3: "STILL"
+4: "UNKNOWN"
+5: "TILTING"
+7: "WALKING"
+8: "RUNNING"
+9: "NONE"
+10: "STOPPED_WHILE_IN_VEHICLE"
+11: "AIR_ON_HSR"
 '''
+def check_mode_from_trip(cleaned_trip, cleaned_sections, section_counter, trip_counter):
+    end_location = cleaned_trip.iloc[trip_counter]["end_loc"]
+    end_loc_lat, end_loc_lon = geojson_to_lat_lon_separated(end_location)
+    modes_from_section = []
+    endsec_location = cleaned_sections.iloc[section_counter]["end_loc"]
+    endsec_loc_lat, endsec_loc_lon = geojson_to_lat_lon_separated(end_sec_location)
+    # Trash value for mode
+    mode = -10
+    if (endsec_loc_lat == end_loc_lat and endsec_loc_lon == end_loc_lon):
+        mode = cleaned_sections.iloc[section_counter]["sensed_mode"]
+        return mode, section_counter + 1
+    while endsec_loc_lat != end_loc_lat and endsec_loc_lon != end_loc_lon and section_counter < len(cleaned_sections) :
+        mode = cleaned_sections.iloc[section_counter]["sensed_mode"]
+        modes_from_section.append(mode)
+        endsec_location = cleaned_sections.iloc[section_counter]["end_loc"]
+        endsec_loc_lon, endsec_loc_lon = geojson_to_lat_lon_separated(endsec_location)
+        if (mode == 0):
+            return mode, section_counter+1
+    return mode, section_counter + 1
```

Comment:
which mode are you returning for the trip? if there is only one mode, you return it, but I am not sure what you are doing with multiple modes?

Reply:
For the modes, I will be redoing the function to determine mode of comparison for the suggestion function and currently am not using the function for the suggestion function.